pkg/gitops/deploykey/github.go

```diff
@@ -0,0 +1,127 @@
+package deploykey
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/google/go-github/v31/github"
+	"github.com/kris-nova/logger"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+	"golang.org/x/oauth2"
+)
+
+const (
+	EnvVarGitHubToken = "GITHUB_TOKEN"
+)
+
+type GitHubProvider struct {
+	cluster     *api.ClusterMeta
+	repoURL     string
+	readOnly    bool
+	githubToken string
+}
+
+func (p *GitHubProvider) Put(ctx context.Context, fluxSSHKey flux.PublicKey) error {
+	owner, repo, ok := p.getOwnerRepoFromRepoURL()
+	if !ok {
+		logger.Info("skipped creating GitHub deploy key from Flux SSH public key for URL %s: Only `git@github.com:OWNER/REPO.git` is accepted for automatic deploy key creation", p.repoURL)
+
+		return nil
+	}
+
+	gh := p.getGitHubAPIClient(ctx)
+
+	logger.Info("creating GitHub deploy key from Flux SSH public key")
+
+	title := p.getDeployKeyTitle()
+
+	key, _, err := gh.Repositories.CreateKey(ctx, owner, repo, &github.Key{
+		Key:      &fluxSSHKey.Key,
+		Title:    &title,
+		ReadOnly: &p.readOnly,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	logger.Info("%s configured with Flux SSH public key\n%s", *key.Title, fluxSSHKey.Key)
+
+	return nil
+}
+
+func (p *GitHubProvider) Delete(ctx context.Context) error {
+	owner, repo, ok := p.getOwnerRepoFromRepoURL()
+	if !ok {
+		logger.Info("skipped deleting GitHub deploy key for URL %s: Only `git@github.com:OWNER/REPO.git` is accepted for automatic deploy key creation", p.repoURL)
+
+		return nil
+	}
+
+	gh := p.getGitHubAPIClient(ctx)
+
+	logger.Info("deleting GitHub deploy key")
+
+	title := p.getDeployKeyTitle()
+
+	keys, _, err := gh.Repositories.ListKeys(ctx, owner, repo, &github.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	var keyID int64
+
+	for _, key := range keys {
+		if key.GetTitle() == title {
+			keyID = key.GetID()
+
+			break
+		}
+	}
+
+	if keyID == 0 {
+		logger.Info("skipped deleting GitHub deploy key %q: The key does not exist. Probably you've already deleted it?")
+
+		return nil
+	}
+
+	if _, err := gh.Repositories.DeleteKey(ctx, owner, repo, keyID); err != nil {
+		return err
+	}
+
+	logger.Info("deleted GitHub deploy key %s", title)
+
+	return nil
+}
+
+func (p *GitHubProvider) getGitHubAPIClient(ctx context.Context) *github.Client {
```

Comment:
Does this matches `ssh://` URLs?

Reply:
No. Should we match `ssh://git@github.com:OWNER/REPO.git`-styled URLs as well?

Reply:
Added support for the full ssh urls in 77ebbd68